### DISCUSSION TITLE
after install SDXL option add FLUX ?

### DISCUSSION THREAD

**Discussion by kalle07** [mcmonkeyprojects/SwarmUI#360] [Q&A] [answered]
iv installed windows one klick install and choose SDXL option

is it that simple for FLUX that i copy a model into model folder 
and than ?
where i put the 3 extra files 
t5xxl
ae
clip_l

and thats all ? 
where i can fast and simple change them eg. for gguf models

and the "Diffusion in Low Bits" option from webui, is the best choice always made automatically?

**Comment by mcmonkey4eva** [comment]
Flux docs here: https://github.com/mcmonkeyprojects/SwarmUI/blob/master/docs/Model%20Support.md#black-forest-labs-flux1-models

GGUF here: https://github.com/mcmonkeyprojects/SwarmUI/blob/master/docs/Model%20Support.md#gguf-quantized-models

you don't need to worry about clip/t5/ae, that's all handled automatically

bitwidth is set correctly automatically, you don't need to worry about that either

**Comment by kalle07** [comment]
omg iam in foocus and webui ... and cant handle that error ^^
(i only changed the model folder, or should flux in a special folder?  all models iv tested same error, all models are working in webui)
SDXL-models works! 
i have choosen my root model folder from webui and leave all other because the same folder structure like yours


errro from fluxmodel
11:48:57.451 [Info] User local requested 1 image with model 'flux1DevFp8_v10.safetensors'...
11:48:59.526 [Warning] [ComfyUI-0/STDERR] !!! Exception during processing !!! ERROR: Could not detect model type of: e:\WebUI_Forge\webui\models\Stable-diffusion\flux1DevFp8_v10.safetensors
11:48:59.527 [Warning] [ComfyUI-0/STDERR] Traceback (most recent call last):
11:48:59.527 [Warning] [ComfyUI-0/STDERR]   File "e:\SwarmUI\dlbackend\comfy\ComfyUI\execution.py", line 323, in execute
11:48:59.528 [Warning] [ComfyUI-0/STDERR]     output_data, output_ui, has_subgraph = get_output_data(obj, input_data_all, execution_block_cb=execution_block_cb, pre_execute_cb=pre_execute_cb)
11:48:59.529 [Warning] [ComfyUI-0/STDERR]                                            ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
11:48:59.529 [Warning] [ComfyUI-0/STDERR]   File "e:\SwarmUI\dlbackend\comfy\ComfyUI\execution.py", line 198, in get_output_data
11:48:59.530 [Warning] [ComfyUI-0/STDERR]     return_values = _map_node_over_list(obj, input_data_all, obj.FUNCTION, allow_interrupt=True, execution_block_cb=execution_block_cb, pre_execute_cb=pre_execute_cb)
11:48:59.530 [Warning] [ComfyUI-0/STDERR]                     ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
11:48:59.530 [Warning] [ComfyUI-0/STDERR]   File "e:\SwarmUI\dlbackend\comfy\ComfyUI\execution.py", line 169, in _map_node_over_list
11:48:59.531 [Warning] [ComfyUI-0/STDERR]     process_inputs(input_dict, i)
11:48:59.531 [Warning] [ComfyUI-0/STDERR]   File "e:\SwarmUI\dlbackend\comfy\ComfyUI\execution.py", line 158, in process_inputs
11:48:59.532 [Warning] [ComfyUI-0/STDERR]     results.append(getattr(obj, func)(**inputs))
11:48:59.532 [Warning] [ComfyUI-0/STDERR]                    ^^^^^^^^^^^^^^^^^^^^^^^^^^^^
11:48:59.532 [Warning] [ComfyUI-0/STDERR]   File "e:\SwarmUI\dlbackend\comfy\ComfyUI\nodes.py", line 543, in load_checkpoint
11:48:59.533 [Warning] [ComfyUI-0/STDERR]     out = comfy.sd.load_checkpoint_guess_config(ckpt_path, output_vae=True, output_clip=True, embedding_directory=folder_paths.get_folder_paths("embeddings"))
11:48:59.533 [Warning] [ComfyUI-0/STDERR]           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
11:48:59.534 [Warning] [ComfyUI-0/STDERR]   File "e:\SwarmUI\dlbackend\comfy\ComfyUI\comfy\sd.py", line 586, in load_checkpoint_guess_config
11:48:59.534 [Warning] [ComfyUI-0/STDERR]     raise RuntimeError("ERROR: Could not detect model type of: {}".format(ckpt_path))
11:48:59.534 [Warning] [ComfyUI-0/STDERR] RuntimeError: ERROR: Could not detect model type of: e:\WebUI_Forge\webui\models\Stable-diffusion\flux1DevFp8_v10.safetensors
11:48:59.535 [Warning] [ComfyUI-0/STDERR]
11:48:59.618 [Error] [BackendHandler] backend #0 failed to load model with error: ComfyUI execution error: ERROR: Could not detect model type of: e:\WebUI_Forge\webui\models\Stable-diffusion\flux1DevFp8_v10.safetensors
11:48:59.619 [Warning] [BackendHandler] backend #0 failed to load model flux1DevFp8_v10.safetensors
11:49:00.482 [Warning] [BackendHandler] All backends failed to load the model 'e:\WebUI_Forge\webui\models\Stable-diffusion/flux1DevFp8_v10.safetensors'! Cannot generate anything.
11:49:00.483 [Error] [BackendHandler] Backend request #3 failed: All available backends failed to load the model 'e:\WebUI_Forge\webui\models\Stable-diffusion/flux1DevFp8_v10.safetensors'.
11:49:00.484 [Error] [BackendHandler] Backend request #3 failed: All available backends failed to load the model 'e:\WebUI_Forge\webui\models\Stable-diffusion/flux1DevFp8_v10.safetensors'.

**Reply by mcmonkey4eva** [reply, marked as answer]
Some flux models require being in the special folder `Models/diffusion_models` instead of the usual folder, specifically any _MMDiT Backbone_ files, including any GGUF. I don't recognize what the file you're using is offhand (looks like a civitai copy?) so it might be a backbone file that needs to be in the special `diffusion_models` folder

**Reply by kalle07** [reply]
okay, yes civitai  ... now it works, thx
